js/src/common/components/AlertManager.js

```diff
@@ -7,20 +7,16 @@ import Alert from './Alert';
  */
 export default class AlertManager extends Component {
   init() {
-    /**
-     * An array of Alert components which are currently showing.
-     *
-     * @type {Alert[]}
-     * @protected
-     */
-    this.components = [];
+    this.state = this.props.state;
   }
 
   view() {
     return (
       <div className="AlertManager">
-        {this.components.map((component) => (
-          <div className="AlertManager-alert">{component}</div>
+        {this.state.activeAlerts.map((alert) => (
+          <div className="AlertManager-alert">
+            <Alert state={alert} ondismiss={this.state.dismiss.bind(this.state, alert.key)} />
```

Comment:
Why not pass the props directly as `<Alert {...props} />` ? That would also remove the need for the additional code inside of `Alert.initProps`.

Reply:
This wouldn't work because we'd need to do a deep clone (when I try, props.children comes up as an empty array), but I can simplify down the initProps